modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -124,6 +126,45 @@ public Product getChannelProduct(
 		return _toProduct(commerceContext, cpDefinition);
 	}
 
+	@Override
+	public Product getChannelProductByFriendlyUrlPath(
+			Long channelId, String friendlyUrl, Long accountId)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannel(channelId);
+
+		Long commerceAccountId = _getCommerceAccountId(
+			accountId, commerceChannel);
+
+		if (!_isAccountEntryEligible(
+				commerceAccountId, commerceChannel.getCommerceChannelId())) {
+
+			return null;
+		}
+
+		Group companyGroup = _groupLocalService.getCompanyGroup(
+			commerceChannel.getCompanyId());
+
+		CPDefinition cpDefinition =
+			_cpDefinitionLocalService.fetchCPDefinitionByFriendlyURL(
+				friendlyUrl, companyGroup.getGroupId());
+
+		if (cpDefinition == null) {
+			throw new NoSuchCProductException();
+		}
```

Comment:
I would use get and not fetch since errors should be handled automatically by the framework, maybe try putting get and looking for a product that doesn't exist